app/core/concurrency.py

```
@@ -0,0 +1,100 @@
+    fraction of 0 or 1 cannot completely lock out either path.
+
+    Args:
+        total: The overall concurrent-request budget.
+        sync_fraction: Fraction of the budget reserved for the sync path.
+
+    Returns:
+        Tuple of (sync capacity, async capacity).
+    """
+    if total <= 0:
+        return 0, 0
+    sync_capacity = max(1, int(round(total * sync_fraction)))
```

Ruff (0.15.14)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)